packages/liveblocks-react-comments/src/primitives/Composer/index.tsx

```diff
@@ -562,7 +562,7 @@ const ComposerSuggestionsListItem = forwardRef<
 const ComposerEditor = forwardRef<HTMLDivElement, ComposerEditorProps>(
   (
     {
-      initialValue,
+      defaultValue,
```

Comment:
I don't think it's an issue to have both depending on their places (controlled/uncontrolled props: `default`, others: `initial`), `initialPresence` does make more sense given the context but I don't know if `initialValue` for an input or `initialCollapsed` here would be worth going against what React does by default (no pun intended 🥁).

Reply:
Perhaps we're entering very subtle territory here, but to me **initial** would represent an initial value to use for a _state_ you can access on a component, whereas a **default** value would provide a fallback value to use in the _absence_ of state (which can happen initially, but also if the real value gets wiped later).

To me the main reason for using **initial** instead of **default** in the RoomProvider was to signify that it would only be used to populate the first ever value, and never beyond that.

This is different from a **default** value. If you wipe the **actual** value, I would expect the default value to show up again. This also means that the default value could be changed and observed over time:

1. Set a default value to "foo" (component shows "foo")
1. Set a **actual** value to "nice" (component shows "nice")
1. Change the default value to "bar" (component shows "nice")
1. Remove the **actual** value (component shows "bar")

Note the component shows "bar" after step (4), not "foo".

If this was named an **initial** value, then I would expect the component to display "" (empty) in step (4) instead. Thus, I'd think of a **default** value a bit like a placeholder — but not in the classic HTML input property `placeholder`, but conceptually similar, but at the React level.

Btw, what's the difference between `defaultValue` and `placeholder` (see comment on L560) here? (Is it an outdated comment?)

So... the tl;dr is that in my mind it's consistent:

- RoomProvider should still use **initial** values
- This component should use **default** value

Reply:
> Btw, what's the difference between defaultValue and placeholder

`defaultValue` is an actual value (e.g. prefilling the composer when editing a comment) but `placeholder` maps to the [placeholder text](https://developer.mozilla.org/en-US/docs/Web/CSS/::placeholder) which doesn't count as a value.

So by your definition `defaultValue` is wrong in the case of `Composer.Editor` because clearing the editor will **not** bring back `defaultValue`, maybe that's why Slate went with `initialValue`. 🤔 Also, like Slate, `Composer.Editor` can only be uncontrolled.

Still, I'm not sure if I want us to do this differently than React given how closely `Composer.Editor` can feel from an `input` or a `textarea`, plus having `initialValue` on one component and `defaultCollapsed` on another.

Reply:
> So by your definition `defaultValue` is wrong in the case of `Composer.Editor` because clearing the editor will **not** bring back `defaultValue`, maybe that's why Slate went with `initialValue`. 🤔

Ah yes, in that case I'd prefer naming this initial value indeed, which is less surprising[^1] I think. I would recommend to stick with `initialValue` then.

> Still, I'm not sure if I want us to do this differently than React given how closely `Composer.Editor` can feel from an `input` or a `textarea`, plus having `initialValue` on one component and `defaultCollapsed` on another.

I get that, and if I hear you correct, it's mostly a case of discoverability. Not sure if it's really going to be an issue, given we have docs, examples, and TypeScript IDE support. If you feel strongly about having the `defaultValue` prop for discoverability, maybe consider putting a `@deprecated` comment on it to point people to the `initialValue` prop instead?

[^1]: I think I once saw a Twitter thread by Dan Abramov on this topic, making a great argument for why they used "initial" instead of "default" in their state hooks, but I can't find it. Maybe it was part of a talk? Not sure. Either way, I think React may have picked `initialValue` if they could.

Reply:
Note: React's `defaultValue` on an `input`/`textarea` also doesn't come back when cleared https://codesandbox.io/s/wizardly-taussig-gxp9ct?file=/src/App.tsx.

To me it's not just about discoverability it's about expectations that come from the ecosystem (largely due to the default React behavior), so even if we think (and maybe React themselves too now) `initial` is the correct one here I'm not sure that's enough to decide to do things differently (speaking purely about interactive components here, I think it's fine for everything else).

Reply:
If you feel strongly about `defaultValue`, specifically to mimic React's `input` naming, I think that's fine 👍